packages/data-broker/src/services/dhis/translators/DhisTranslator.ts

```diff
@@ -221,10 +221,19 @@ export class DhisTranslator {
     metadata.forEach(mData => {
       const translatedMetadata = dataElements
         .filter(dataSource => dataSource.dataElementCode === mData.code)
-        .map(({ code, dataElementCode }) => ({
-          ...metadataGroupedByCode[dataElementCode],
-          code,
-        }));
+        .map(({ code, dataElementCode, config }) => {
+          const { id, name } = metadataGroupedByCode[dataElementCode];
+          const { categoryOptionCombo: categoryOptionComboCode } = config;
+          const categoryOptionCombo = categoryOptionCombos.find(
+            item => item.code === categoryOptionComboCode,
+          );
+          const dataElementName = formatInboundDataElementName(name, categoryOptionCombo?.name);
```

Comment:
Now the formatting for the data element name matches what comes back with the analytics `metadata` object.